---
title: "Intro 1: Download GenBank"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Intro 1: Download GenBank}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

In the first of the restez introductory tutorials we will explain how to download parts of GenBank on to your computer and from these downloads, construct a local database. GenBank can be downloaded from NCBI using the File Transfer Protocol (FTP), the online portal for these downloads can be viewed from your web browser,
[FTP NCBI GenBank](ftp://ftp.ncbi.nlm.nih.gov/genbank/). GenBank is hosted at this portal as 1000s of compressed sequence files called flatfiles. New versions of these flatfiles are released once a month as new sequence information is uploaded. For information on the latest GenBank release, see the [NCBI statistics page](https://www.ncbi.nlm.nih.gov/genbank/statistics/).

Restez downloads a selection of these flatfiles, decompresses them and unpacks them into an SQLite database that can be searched using its functions. Because there are potentially huge amounts of sequence information, all of which is unlikely to be of interest to a user, restez allows you to select the sets of flatfiles to download. For the most part these sets of flatfile types are determined by taxonomy: a user can download all primates or plant sequences, for example.

Here, we will explain how to set up restez, download genbank and then create a database from the downloaded files.

## Setting up

Before we can download, we need to tell restez where we would like to store the downloaded files and the eventual GenBank database. To do this we use the `restez_path_set` function. A user provides a system file path to a folder, and within that folder a new folder called 'restez' is created. In this example, I will create a restez folder in my desktop.

```{r setting up, eval=FALSE, include=TRUE}
library(restez)
# On my linux computer and I specify my desktop
fp <- '~/Desktop'
restez_path_set(filepath = fp)
```
```
... Creating '~/Desktop/restez'
... Creating '~/Desktop/restez/downloads'
```

Messages are printed to console telling us that the path was created and that a downloads subfolder was created.

> **Note**, whenever you intend to use restez you will need to specify the restez path. If you are using it regularly, it might be a good idea to set it using your .Rprofile.

## Download

Next, we download genbank using `gb_download()`. This is an interactive function, it looks up the latest GenBank release, parses the release notes and prints to console the available sets of flatfiles that can be downloaded.

```{r download, eval=FALSE, include=TRUE}
gb_download()
```
```
──────────────────────────────────────────────────────────────────────────────────────────────────────
Looking up latest GenBank release ...
... found release 224
Found 3057 sequence files
──────────────────────────────────────────────────────────────────────────────────────────────────────
Which sequence file types would you like to download?
Choose from those listed below:
● 1  -  'EST (expressed sequence tag)' (485 files available)
● 2  -  'Bacterial' (450 files available)
● 3  -  'Constructed' (362 files available)
● 4  -  'Patent' (323 files available)
● 5  -  'GSS (genome survey sequence)' (304 files available)
● 6  -  'TSA (transcriptome shotgun assembly)' (234 files available)
● 7  -  'Plant sequence entries (including fungi and algae),' (168 files available)
● 8  -  'Invertebrate' (161 files available)
● 9  -  'HTGS (high throughput genomic sequencing)' (154 files available)
● 10  -  'Environmental sampling' (101 files available)
● 11  -  'Other vertebrate' (85 files available)
● 12  -  'Primate' (58 files available)
● 13  -  'Viral' (53 files available)
● 14  -  'Other mammalian' (39 files available)
● 15  -  'Rodent' (30 files available)
● 16  -  'STS (sequence tagged site)' (20 files available)
● 17  -  'HTC (high throughput cDNA sequencing)' (15 files available)
● 18  -  'Synthetic and chimeric' (10 files available)
● 19  -  'Phage' (4 files available)
● 20  -  'Unannotated' (1 files available)
Provide one or more numbers separated by spaces.
e.g. to download all Mammal sequences type: "12 14 15" followed by Enter
Which files would you like to download?
(Press Esc to quit) 
```

We are prompted to provide a number indicating which set we would like to download. In the examples of these tutorials we make use of one of the smaller sets, the rodents, which is number 15 in this example.

```
(Press Esc to quit) 15
You've selected a total of 30 file types. These represent:
● 'Rodent'
Each file contains about 250 MB of decompressed data.
30 files amounts to about 7.5 GB
Additionally, the resulting SQL database takes about 500 MB per file
In total, the uncompressed files and SQL database should amount to 22.5 GB Is that OK?
Enter any key to continue or press Esc to quit 
──────────────────────────────────────────────────────────────────────────────────────────────────────
Downloading ...
... 'gbrod1.seq' (1 / 30)
```

Now the download has begun, we just need to wait for it to finish. This can take quite a while.

## Create a database

After the download has completed, we need to use `db_create()` to create our local database. This looks in the downloads folder of our restez path, decompresses any .seq.tar files, breaks these files up into separate GenBank records and adds them to an SQLite database.

```{r create database, eval=FALSE, include=TRUE}
db_create()
```
```
Decompressing and adding 30 files to database ...
... 'gbrod10.seq.gz'(1 / 30)
... ... decompressing
... ... adding
... 'gbrod11.seq.gz'(2 / 30)
... ... decompressing
... ... adding
```

Again, this can take quite a while.

## Checking the setup

After the download and the database creation steps are complete, we can confirm our setup with the following commands.

```{r confirm setup, eval=FALSE, include=TRUE}
(restez_path_get())
status_check()
```
```
[1] "~/Desktop/restez"
Checking setup status ...
... found 31 files in 'downloads/'
... totalling 7.28 GB
... found 'sql_db' of 12.23 GB
```

The above messages tell us that the restez path is set to the desktop and that the restez folder contains a database of 12.23 GB and that the downloads folder contains all of the rodent downloads, 7.28 GB.

If your status shows the same, then you are ready to move to the next tutorial, Intro 2: Querying with restez.
