<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<ol start="2" style="list-style-type: decimal">
<li>How to search for and fetch sequences</li>
</ol> • restez</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;ol start=" style="list-style-type: decimal">
</head>
<body>
<li>How to search for and fetch sequences</li>
" /&gt;

<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_rodents.html">1. Build a database of all rodents</a>
    </li>
    <li>
      <a href="../articles/2_search_and_fetch.html">2. How to search for and fetch sequences</a>
    </li>
    <li>
      <a href="../articles/3_parsing.html">3. Advanced parsing of a GenBank record</a>
    </li>
    <li>
      <a href="../articles/4_phylotar.html">4. Running phylotaR with restez</a>
    </li>
    <li>
      <a href="../articles/5_tips_and_tricks.html">5. Tips and Tricks</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AntonelliLab/restez#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1><ol start="2" style="list-style-type: decimal">
<li>How to search for and fetch sequences</li>
</ol></h1>
            
            <h4 class="date">2018-09-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/AntonelliLab/restez#readme/blob/master/vignettes/2_search_and_fetch.Rmd"><code>vignettes/2_search_and_fetch.Rmd</code></a></small>
      <div class="hidden name"><code>2_search_and_fetch.Rmd</code></div>

    </div>

    
    
<p>A downside of the <code>restez</code> approach is we are unable to then search the local database for sequences of interest as the database cannot possibly contain sufficient metadata (most notably taxonomic information) to perform as rich a search as online via NCBI. As a result we must perform the sequence discovery process independently from <code>restez</code>. In this tutorial we will demonstrate how to peform online searches using the <code>rentrez</code> package and then use the search results to look up sequences in a <code>restez</code> database.</p>
<p>To run all the code in this tutorial you will need to have already set up the rodents database, see <a href="https://antonellilab.github.io/restez/articles/rodents.html">Build a database of all rodents</a>.</p>
<div id="search-ncbi" class="section level2">
<h2 class="hasAnchor">
<a href="#search-ncbi" class="anchor"></a>Search NCBI</h2>
<p>Let’s pretend we’re interested in COI sequences for a group of rodents called the Sciuromorpha. We can create an NCBI search term; use <code>entrez_search</code> to perform the search; and then retrieve the accession IDs using <code>entrez_fetch</code> with the <code>acc</code> rettype. <code>entrez_fetch</code> returns a single text that will need to be split up by the newline character. Finally, we should drop any version number after the downloaded accessions. This is not strictly necessary, but it makes it easier to check our results from <code>restez</code> later.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># 33553 - Sciuromorpha - squirrel-like things</span>
search_term &lt;-<span class="st"> 'txid33553[Organism:exp] AND COI [GENE] AND 100:1000[SLEN]'</span>
search_object &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">term =</span> search_term,
                                        <span class="dt">use_history =</span> <span class="ot">TRUE</span>, <span class="dt">retmax =</span> <span class="dv">0</span>)
accessions &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                    <span class="dt">web_history =</span> search_object<span class="op">$</span>web_history,
                                    <span class="dt">rettype =</span> <span class="st">'acc'</span>)
accessions &lt;-<span class="st"> </span><span class="kw">strsplit</span>(<span class="dt">x =</span> accessions, <span class="dt">split =</span> <span class="st">'</span><span class="ch">\\</span><span class="st">n'</span>)[[<span class="dv">1</span>]]
accessions &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">'</span><span class="ch">\\</span><span class="st">.[0-9]+'</span>, <span class="dt">replacement =</span> <span class="st">''</span>, <span class="dt">x =</span> accessions)
<span class="kw">print</span>(<span class="kw">length</span>(accessions))
<span class="co">#&gt; [1] 456</span>
<span class="kw">print</span>(accessions[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])
<span class="co">#&gt;  [1] "JF457164" "JF457162" "JF457161" "JF457159" "JF457156" "JF457155"</span>
<span class="co">#&gt;  [7] "JF457154" "JF457153" "JF457152" "JF457151"</span></code></pre></div>
</div>
<div id="retrieve-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-sequences" class="anchor"></a>Retrieve sequences</h2>
<p>To fetch the sequences from the rodents database, we can just use the <code>gb_fasta_get</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; restez v0.1.0</span>
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; Remember to restez_path_set() and, then, restez_connect()</span>
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rodents_path)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
coi_sequences &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span>(<span class="dt">id =</span> accessions)
<span class="kw">str</span>(coi_sequences[[<span class="dv">1</span>]])
<span class="co">#&gt;  chr "&gt;FJ808614.1 Glis glis isolate A1g02_E11-F cytochrome oxidase subunit I (COI) gene, partial cds; mitochondrial\n"| __truncated__</span>
<span class="co"># Are all accessions in results?</span>
<span class="kw">all</span>(accessions <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(coi_sequences))
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>
<div id="comparing-to-entrez" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-to-entrez" class="anchor"></a>Comparing to Entrez</h2>
<p>Can we not just use <code>rentrez</code> to do the fetching as well? Yes, but <code>restez</code> can be a lot faster. NCBI limits the number of requests per user, often to as little as 100 items per request with varying time delays. Additionally for any programmatic retrieval of sequences using an online server can never be as reliable as a local copy.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># time via restez</span>
<span class="kw">system.time</span>(<span class="dt">expr =</span> {
  coi_sequences &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span>(<span class="dt">id =</span> accessions)
  })
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   1.011   2.006   4.570</span>
<span class="co"># time via Entrez</span>
<span class="kw">system.time</span>(<span class="dt">expr =</span> {
  coi_sequences_p1 &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                            <span class="dt">id =</span> accessions[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>],
                                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
  coi_sequences_p2 &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                            <span class="dt">id =</span> accessions[<span class="dv">101</span><span class="op">:</span><span class="dv">200</span>],
                                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
  coi_sequences_p3 &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                            <span class="dt">id =</span> accessions[<span class="dv">201</span><span class="op">:</span><span class="dv">300</span>],
                                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
  coi_sequences_p4 &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                            <span class="dt">id =</span> accessions[<span class="dv">301</span><span class="op">:</span><span class="dv">400</span>],
                                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
  coi_sequences_p5 &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                            <span class="dt">id =</span> accessions[<span class="dv">401</span><span class="op">:</span><span class="dv">456</span>],
                                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
  })
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   0.131   0.014   4.978</span>
<span class="co"># always disconnect</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
<!-- Below is no longer relevant now that the size of sequences in the db has been limited.
##Missing

A user should know that if an ID cannot be found in the local database no error or warning is raised. This is why it can be good practice to test whether all the provided IDs are in the returned named vector. In this example, we can see that not all the accession IDs that were provided are in the returned `coi_sequences`. Why is that?


```r
# Are all accessions in results?
all(accessions %in% names(coi_sequences))
#> [1] TRUE
# .... no
```

This is because `restez` only downloads GenBank sequences, and these 'missing' sequence IDs are RefSeq sequences. GenBank, however, acts as the source database for RefSeq -- which is the case for other NCBI databases too -- and all the missing RefSeq sequences can also be found in GenBank under a different ID.

We can look up alternative IDs and test whether they are in our `accesssions` vector, so:


```r
(accessions[!accessions %in% names(coi_sequences)])
#> character(0)
# NC* refers to RefSeq sequences and are not available through restez
# The sequence exists in GB under a different id which we can find like so
smmry <- rentrez::entrez_summary(db = 'nucleotide', id = 'NC_027278')
# This ID does exist in our results.
(smmry$assemblyacc %in% accessions)
#> [1] FALSE
```
-->
</div>
<div id="next-up" class="section level2">
<h2 class="hasAnchor">
<a href="#next-up" class="anchor"></a>Next up</h2>
<p><strong><a href="https://antonellilab.github.io/restez/articles/parsing.html">Advanced parsing of a GenBank record</a></strong></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#search-ncbi">Search NCBI</a></li>
      <li><a href="#retrieve-sequences">Retrieve sequences</a></li>
      <li><a href="#comparing-to-entrez">Comparing to Entrez</a></li>
      <li><a href="#next-up">Next up</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
