<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<ol start="3" style="list-style-type: decimal">
<li>Advanced parsing of a GenBank record</li>
</ol> • restez</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;ol start=" style="list-style-type: decimal">
</head>
<body>
<li>Advanced parsing of a GenBank record</li>
" /&gt;

<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_rodents.html">1. Build a database of all rodents</a>
    </li>
    <li>
      <a href="../articles/2_search_and_fetch.html">2. How to search for and fetch sequences</a>
    </li>
    <li>
      <a href="../articles/3_parsing.html">3. Advanced parsing of a GenBank record</a>
    </li>
    <li>
      <a href="../articles/4_phylotar.html">4. Running phylotaR with restez</a>
    </li>
    <li>
      <a href="../articles/5_tips_and_tricks.html">5. Tips and Tricks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/restez">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1><ol start="3" style="list-style-type: decimal">
<li>Advanced parsing of a GenBank record</li>
</ol></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/restez/blob/master/vignettes/3_parsing.Rmd"><code>vignettes/3_parsing.Rmd</code></a></small>
      <div class="hidden name"><code>3_parsing.Rmd</code></div>

    </div>

    
    
<p>In this tutorial we will demonstrate how the <code><a href="../reference/gb_extract.html">gb_extract()</a></code> function works. <code>restez</code> downloads and stores all GenBank records in text format. Ordinarily, to be able to extract relevant bits of information from a text record in a systematic way we would need to make use of <a href="https://en.wikipedia.org/wiki/Regular_expression">regular expressions</a>. <code><a href="../reference/gb_extract.html">gb_extract()</a></code> uses regular expressions, so we don’t have to. Here’s how it works.</p>
<div id="a-genbank-record" class="section level2">
<h2 class="hasAnchor">
<a href="#a-genbank-record" class="anchor"></a>A GenBank record</h2>
<p>restez comes with an example GenBank record, AY952423, which can be viewed <a href="https://www.ncbi.nlm.nih.gov/nuccore/AY952423.1">online</a>. We can retrieve the record with <code>rentrez</code> or load it using the <code><a href="https://www.rdocumentation.org/packages/utils/topics/data">data()</a></code>. We can visualise the record in R as it appears online using <code><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat()</a></code> – like print but with newline spaces parsed correctly.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(restez)
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; restez v1.0.0</span>
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; Remember to restez_path_set() and, then, restez_connect()</span>
<span class="co"># record &lt;- rentrez::entrez_fetch(db = 'nucleotide', id = 'AY952423', rettype = 'gb', retmode = 'text')</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(record)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(record)
<span class="co">#&gt; LOCUS       AY952423                2623 bp    DNA     linear   PLN 17-APR-2005</span>
<span class="co">#&gt; DEFINITION  Livistona chinensis tRNA-Lys (trnK) gene, partial sequence; and</span>
<span class="co">#&gt;             matK gene, complete sequence; chloroplast.</span>
<span class="co">#&gt; ACCESSION   AY952423</span>
<span class="co">#&gt; VERSION     AY952423.1</span>
<span class="co">#&gt; KEYWORDS    .</span>
<span class="co">#&gt; SOURCE      chloroplast Livistona chinensis</span>
<span class="co">#&gt;   ORGANISM  Livistona chinensis</span>
<span class="co">#&gt;             Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;</span>
<span class="co">#&gt;             Spermatophyta; Magnoliophyta; Liliopsida; Arecaceae; Coryphoideae;</span>
<span class="co">#&gt;             Livistoneae; Livistoninae; Livistona.</span>
<span class="co">#&gt; REFERENCE   1  (bases 1 to 2623)</span>
<span class="co">#&gt;   AUTHORS   Li,X.X. and Zhou,Z.K.</span>
<span class="co">#&gt;   TITLE     Monocotyledons phylogeny based on three genes (matK, rbcL and 18S</span>
<span class="co">#&gt;             rDNA) sequences</span>
<span class="co">#&gt;   JOURNAL   Unpublished</span>
<span class="co">#&gt; REFERENCE   2  (bases 1 to 2623)</span>
<span class="co">#&gt;   AUTHORS   Li,X.X. and Zhou,Z.K.</span>
<span class="co">#&gt;   TITLE     Direct Submission</span>
<span class="co">#&gt;   JOURNAL   Submitted (03-MAR-2005) Taxonomical and Ethnobotanical Department,</span>
<span class="co">#&gt;             Kunming Institute of Botany, The Chinese Academy of Sciences,</span>
<span class="co">#&gt;             Heilongtan, Kunming, Yunnan 650204, China</span>
<span class="co">#&gt; FEATURES             Location/Qualifiers</span>
<span class="co">#&gt;      source          1..2623</span>
<span class="co">#&gt;                      /organism="Livistona chinensis"</span>
<span class="co">#&gt;                      /organelle="plastid:chloroplast"</span>
<span class="co">#&gt;                      /mol_type="genomic DNA"</span>
<span class="co">#&gt;                      /db_xref="taxon:115492"</span>
<span class="co">#&gt;      gene            &lt;1..&gt;2623</span>
<span class="co">#&gt;                      /gene="trnK"</span>
<span class="co">#&gt;                      /note="tRNA-Lys"</span>
<span class="co">#&gt;      intron          &lt;1..&gt;2623</span>
<span class="co">#&gt;                      /gene="trnK"</span>
<span class="co">#&gt;      gene            813..2347</span>
<span class="co">#&gt;                      /gene="matK"</span>
<span class="co">#&gt;      misc_feature    813..2347</span>
<span class="co">#&gt;                      /gene="matK"</span>
<span class="co">#&gt;                      /note="similar to maturase K"</span>
<span class="co">#&gt; ORIGIN      </span>
<span class="co">#&gt;         1 attggggttg ctaactcaac ggtagagtac tcggctttta agtgcgacta tcatctttta</span>
<span class="co">#&gt;        61 cacatttgga tgaagtaagg aattcgtcca gactattggt agagtctata agaccacgac</span>
<span class="co">#&gt;       121 tgatcctgaa aggtaatgaa tggaaaaaat agcatgtcgt acgtaataca atgagaaact</span>
<span class="co">#&gt;       181 tgtaatttct tattgtaatt ttttaagtag aactttgagt ttatccttac tggatcatta</span>
<span class="co">#&gt;       241 caaaaatatt gtattttatt tttggaaggg gacgaaaaaa aggaaattcc caacatttat</span>
<span class="co">#&gt;       301 tgtttggtct aatgaataaa tggatagggg cctagggtag ggcccaattt ttgtaaaaca</span>
<span class="co">#&gt;       361 aaaagcaacg agcttatgtt cttaatttga ataattaccc gatctaatta gatgttaaaa</span>
<span class="co">#&gt;       421 ataaattagt gccagatgtg gtaaagggtt ctactgtaag tggacctttt tttttttttt</span>
<span class="co">#&gt;       481 ttatgaatcc tacctattat ctattatgga ttaaagatgg atgtgtataa gaagaagtat</span>
<span class="co">#&gt;       541 actgataaag agaatttttc caaagtcaaa agagcaatcg ggttgcaaaa ataaaggatt</span>
<span class="co">#&gt;       601 tttacctccg agaattataa attaattgga tcaaaaggag aggaaaaagt ctgtgattgg</span>
<span class="co">#&gt;       661 actccttcta tccgcgggta tgggtatata gtaggtatat atgtatattt gtatactata</span>
<span class="co">#&gt;       721 taaattacat gccctgttct gaccgtattg cactatgtat tatttgataa tccaagaaat</span>
<span class="co">#&gt;       781 gcctcctact tctggttcaa gtagaaatga aaatagaaga attacaagaa tatttagaaa</span>
<span class="co">#&gt;       841 aagatagatc tcggcaacaa cacttcctat acccactttt ctttcaggag tatatttatg</span>
<span class="co">#&gt;       901 cacttgctca tgattatggg tttaaagggt tcgatttttt acgaacctat ggaaattggg</span>
<span class="co">#&gt;       961 ggttatgata ataaatctag ttcagtactt gtaaaacatt taattactcg aatgtatcaa</span>
<span class="co">#&gt;      1021 cagaattatt tgatttattc tgttaatgaa tctaaccaaa atcgattgat tgagcataac</span>
<span class="co">#&gt;      1081 aattcttttt attctcaaat gatatctgaa gtttttgcga tcattgcaga aattccattc</span>
<span class="co">#&gt;      1141 tctcagcaat tactattttc tcttcgagga aaaaagaata ccaaaatctc agactttacg</span>
<span class="co">#&gt;      1201 atctattcat tcaatatttc cctttttaga agacaaatta tcacatttaa actatgtgtc</span>
<span class="co">#&gt;      1261 agatatatta ataccctatc ccatccattt ggaaatcttg gtgcaaattc ttcaatgctg</span>
<span class="co">#&gt;      1321 gatccaagat gtttcttctt tgcatttatt gcgattcttt ctccacgaac atcataatgg</span>
<span class="co">#&gt;      1381 gaatagtttt ttttttccaa agaaatcctt ttcaaaagaa aataaaagac tctttcgatt</span>
<span class="co">#&gt;      1441 cctatataat tcttatgtat ctgaatgtga atttgtctta gtgtttcttc gtaaacaatc</span>
<span class="co">#&gt;      1501 ctcttattta caatcaaaat cctatggaat ctttcttgag cgaacacatt tctatggaag</span>
<span class="co">#&gt;      1561 aatggaacat cttatagtag tgtgtcataa ttattgtcag aaggcctttt gggtcttcaa</span>
<span class="co">#&gt;      1621 ggatcctttt atgcattatg ttcgatatca aggaaaagca attctggcat caaaaggatc</span>
<span class="co">#&gt;      1681 ttatcttttg atgaagaaat ggagatgtca tcttgtcaat ttctggcaat attattttca</span>
<span class="co">#&gt;      1741 tttttgggct cagccttaca gaatttcaat aaaccaatta ggaaatcatt ccttctattt</span>
<span class="co">#&gt;      1801 tctcggttat ctttcaagtg tattaaaaaa tacttcgtct gtaaggaatc aaatgctaga</span>
<span class="co">#&gt;      1861 gaattccttt ttaatagata ctattactaa taaattggat accatagtcc cagttcttcc</span>
<span class="co">#&gt;      1921 tcttattgga tctttgtcta aagctaaatt ttgtaccgta tccgggcatc ctagtagtaa</span>
<span class="co">#&gt;      1981 gccaatctgg acggatttat cggattctga tattattgat agatttggtc ggatatgtag</span>
<span class="co">#&gt;      2041 aaatctttct cattattata gtggatcctc aaaaaaacag agcttatatc gaataaggta</span>
<span class="co">#&gt;      2101 tatacttcga ctttcttgtg ctagaacttt agctcgtaaa cataaaagta cagtacgtgc</span>
<span class="co">#&gt;      2161 ttttttgcaa agattaggtt cggaattatt agaagaattc tttacagaag aagaaggagt</span>
<span class="co">#&gt;      2221 tgtttttttg atttcccaaa agaacaaaac ctcttttcct ctctataggt cacatagaga</span>
<span class="co">#&gt;      2281 acgcatttgg tatttggata ttatccatat taatgaattg gtgaattcat ttatgatggg</span>
<span class="co">#&gt;      2341 gcgataagcc cctataaaat aagaaatata aattttttct aatgtctaat aaatagacga</span>
<span class="co">#&gt;      2401 caaattcatt aattttcatt ctgaaatgct catctagtag tgtagtgatt gaatcaactg</span>
<span class="co">#&gt;      2461 agtattcaaa atttttagac aaacttctag ggatagaagt ttgttttatc tgtatacata</span>
<span class="co">#&gt;      2521 ggtaaagtcg tgtgcaatga aaaatgcaag cacgatttgg ggagagataa ttttctctat</span>
<span class="co">#&gt;      2581 tgtaacaaat aaaaattatc tactccatcc gactagttaa tcg</span>
<span class="co">#&gt; //</span></code></pre></div>
</div>
<div id="extracting" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting" class="anchor"></a>Extracting</h2>
<p>We can extract different elements of the above record with <code><a href="../reference/gb_extract.html">gb_extract()</a></code> ….</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># such as the LOCUS information ...</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'locus'</span>))
<span class="co">#&gt;     accession        length           mol          type        domain </span>
<span class="co">#&gt;    "AY952423"        "2623"         "DNA"      "linear"         "PLN" </span>
<span class="co">#&gt;          date </span>
<span class="co">#&gt; "17-APR-2005"</span>
<span class="co"># the accession</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'accession'</span>))
<span class="co">#&gt; [1] "AY952423"</span>
<span class="co"># the accession + version</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'version'</span>))
<span class="co">#&gt; [1] "AY952423.1"</span>
<span class="co"># the organism name</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'organism'</span>))
<span class="co">#&gt; [1] "Livistona chinensis"</span>
<span class="co"># the sequence definition line</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'definition'</span>))
<span class="co">#&gt; [1] "Livistona chinensis tRNA-Lys (trnK) gene, partial sequence; and matK gene, complete sequence; chloroplast"</span>
<span class="co"># the keywords (this record doesn't have any ....)</span>
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'keywords'</span>))
<span class="co">#&gt; character(0)</span>
<span class="co"># even the features as a list object</span>
features &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'features'</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(features[[<span class="dv">1</span>]])
<span class="co">#&gt; $type</span>
<span class="co">#&gt; [1] "source"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $location</span>
<span class="co">#&gt; [1] "1..2623"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $organism</span>
<span class="co">#&gt; [1] "Livistona chinensis"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $organelle</span>
<span class="co">#&gt; [1] "plastid:chloroplast"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $mol_type</span>
<span class="co">#&gt; [1] "genomic DNA"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $db_xref</span>
<span class="co">#&gt; [1] "taxon:115492gene            &lt;1..&gt;2623"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gene</span>
<span class="co">#&gt; [1] "trnK"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $note</span>
<span class="co">#&gt; [1] "tRNA-Lysintron          &lt;1..&gt;2623"</span>
<span class="co"># and of course the sequence itself</span>
seq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'sequence'</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/str">str</a></span>(seq)
<span class="co">#&gt;  chr "ATTGGGGTTGCTAACTCAACGGTAGAGTACTCGGCTTTTAAGTGCGACTATCATCTTTTACACATTTGGATGAAGTAAGGAATTCGTCCAGACTATTGGTAGAGTCTATAA"| __truncated__</span></code></pre></div>
</div>
<div id="from-the-database" class="section level2">
<h2 class="hasAnchor">
<a href="#from-the-database" class="anchor"></a>From the database</h2>
<p>You can try out the above functions yourself on any sequence record by downloading them through the <a href="https://github.com/ropensci/rentrez"><code>rentrez</code> package</a> using <code><a href="../reference/entrez_fetch.html">entrez_fetch(db='nucleotide', rettype='gb')</a></code>. Or why not test them out using any of the records from the rodents database?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(restez)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rodents_path)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
(rand_id &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/warning">suppressWarnings</a></span>(<span class="kw"><a href="../reference/list_db_ids.html">list_db_ids</a></span>()), <span class="dv">1</span>))
<span class="co">#&gt; [1] "AB006830"</span>
record &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_record_get.html">gb_record_get</a></span>(rand_id)
(<span class="kw"><a href="../reference/gb_extract.html">gb_extract</a></span>(<span class="dt">record =</span> record, <span class="dt">what =</span> <span class="st">'features'</span>))
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; [[1]]$type</span>
<span class="co">#&gt; [1] "source"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$location</span>
<span class="co">#&gt; [1] "1..369"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$organism</span>
<span class="co">#&gt; [1] "Mus sp."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$mol_type</span>
<span class="co">#&gt; [1] "mRNA"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$db_xref</span>
<span class="co">#&gt; [1] "IMGT/LIGM:AB006830"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$cell_line</span>
<span class="co">#&gt; [1] "APU-6"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$cell_type</span>
<span class="co">#&gt; [1] "hybridomaCDS             &lt;1..&gt;369"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$codon_start</span>
<span class="co">#&gt; [1] "1"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$product</span>
<span class="co">#&gt; [1] "anti-pseudouridine monoclonal antibody heavychain variable region"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$protein_id</span>
<span class="co">#&gt; [1] "BAA22169.1"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]]$translation</span>
<span class="co">#&gt; [1] "QVKLLESGPELGKPGASMKISCKVCGYSFTAYTMNWVKQSHGKTLEWIGLINLFNGFTLYNQKFKGKATLTVDRSSTTAYMEILSLTSDDSAVYYCARSDYYGSSSPYYFDFWGQGTTLTVSSV_region        &lt;1..&gt;369"</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
</div>
<div id="next-up" class="section level2">
<h2 class="hasAnchor">
<a href="#next-up" class="anchor"></a>Next up</h2>
<p><strong><a href="https://ropensci.github.io/restez/articles/4_phylotar.html">Running phylotaR with restez</a></strong></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#a-genbank-record">A GenBank record</a></li>
      <li><a href="#extracting">Extracting</a></li>
      <li><a href="#from-the-database">From the database</a></li>
      <li><a href="#next-up">Next up</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
