<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>phylotaR • restez</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="phylotaR">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/parsing.html">GenBank record parsing</a>
    </li>
    <li>
      <a href="../articles/phylotar.html">phylotaR</a>
    </li>
    <li>
      <a href="../articles/search_and_fetch.html">Search and Fetch</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AntonelliLab/restez#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>phylotaR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AntonelliLab/restez#readme/blob/master/vignettes/phylotar.Rmd"><code>vignettes/phylotar.Rmd</code></a></small>
      <div class="hidden name"><code>phylotar.Rmd</code></div>

    </div>

    
    
<div id="intoduction" class="section level1">
<h1 class="hasAnchor">
<a href="#intoduction" class="anchor"></a>Intoduction</h1>
<p>In this demonstration we will showcase how a <code>restez</code> database can be used to speed up a <a href="https://github.com/AntonelliLab/phylotaR">phylotaR</a> run. phylotaR runs an automated pipeline for identifying ortholgous gene clusters as the first step in a phylogenetic analysis. A user provides a taxonomic identity and the pipeline downloads all relevant sequences and identifies clusters using a local-alignment search tool. For more information on phylotaR see its <a href="https://doi.org/10.3390/life8020020">published article</a>.</p>
<p>By using restez in conjunction with phylotaR, we will not only being saving time, but also improving the chances of a successful phylotaR run – often NCBI Entrez limits the number of requests or even rejects requests from IP addresses that are making too many. Note, however, that the gains in using restez with phylotaR only make sense if you make use of the restez database multiple times or if you wish to radically increase the maximum number of sequences to download per taxon (by default it is only 3,000). Also, note that using a restez database does not currently eliminate the need for an internet connection. phylotaR still needs to look up taxonomic information and must also identify relevant sequence IDs using Entrez (this may change in the future as restez develops).</p>
<p>We will run a phylotaR run for the beaver-like, rodent suborder, Caviomorpha, the clade that contains things like beavers and gophers. We will first set-up our restez database followed by our phylotaR pipeline and then run the pipeline. To run the entire pipeline can take a long time (1 or 2 hours, with most time being taken by the database set-up) but depends on your internet connection and computer specs.</p>
</div>
<div id="restez-set-up" class="section level1">
<h1 class="hasAnchor">
<a href="#restez-set-up" class="anchor"></a>Restez set-up</h1>
<p>First we need to create a local GenBank database. Because the Caviomorpha are rodents we must select the rodent GenBank files. We also should limit the size of the resulting SQL database by specifying minimum and maximum sequence lengths. Making requests from a large SQL database can take a long time to complete, limiting the advantages of generating a restez database. Since in our phylotaR pipeline we are only interested in sequences of greater than 250 bp but less than 2000 bp, we can limit the restez database size at its inception. The <code><a href="../reference/db_create.html">db_create()</a></code> function can be run again later using the same downloaded files if you would like to change the minimum and maximum sequence lengths contained within it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)

<span class="co"># Setup</span>
restez_path &lt;-<span class="st"> '[PATH/WHERE/TO/STORE/DATABASE]'</span>
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(<span class="dt">filepath =</span> restez_path)

<span class="co"># Download</span>
<span class="kw"><a href="../reference/db_download.html">db_download</a></span>()  <span class="co"># select Rodents, in this run it was 15.</span>

<span class="co"># Create</span>
<span class="co"># only build a database of relevant sequences by limiting sequence length</span>
<span class="kw"><a href="../reference/db_create.html">db_create</a></span>(<span class="dt">db_type =</span> <span class="st">'nucleotide'</span>, <span class="dt">min_length =</span> <span class="dv">250</span>, <span class="dt">min_length =</span> <span class="dv">2000</span>)

<span class="co"># Did it download correclty?</span>
<span class="kw"><a href="../reference/restez_status.html">restez_status</a></span>()</code></pre></div>
</div>
<div id="phylotar-run" class="section level1">
<h1 class="hasAnchor">
<a href="#phylotar-run" class="anchor"></a>phylotaR run</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Currently only a specific version of phylotaR can work with restez</span>
<span class="co"># It must be installed from GitHub using devtools, restez branch</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="dt">repo =</span> <span class="st">'AntonelliLab/phylotaR'</span>, <span class="dt">ref =</span> <span class="st">'restez'</span>)</code></pre></div>
<p>Now we have set-up a restez database for rodents, we can run our phylotaR pipeline. We do not need to set-up the pipeline any differently with a restez database, except to ensure we have called the restez library and pointed the package to the rodent database.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phylotaR)
<span class="kw">library</span>(restez)

<span class="co"># Restez</span>
restez_path &lt;-<span class="st"> '[PATH/WHERE/TO/RODENT/DATABASE]'</span>
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(<span class="dt">filepath =</span> restez_path)

<span class="co"># Vars</span>
wd &lt;-<span class="st"> 'beavers'</span>
txid &lt;-<span class="st"> </span><span class="dv">1963757</span>  <span class="co"># Castorimorpha</span>
v &lt;-<span class="st"> </span><span class="ot">TRUE</span>        <span class="co"># print progress to console</span>
btchsz &lt;-<span class="st"> </span><span class="dv">100</span>    <span class="co"># max number of downloads per request</span>
mnsql &lt;-<span class="st"> </span><span class="dv">250</span>     <span class="co"># minimum sequence length</span>
mxsql &lt;-<span class="st"> </span><span class="dv">2000</span>    <span class="co"># maximum sequence length</span>
ncbi_dr &lt;-<span class="st"> '[PATH/TO/BLAST]'</span> <span class="co"># e.g. '/usr/local/ncbi/blast/bin'</span>

<span class="co"># Setup and run</span>
<span class="kw">dir.create</span>(wd)
<span class="kw"><a href="../reference/setup.html">setup</a></span>(<span class="dt">wd =</span> wd, <span class="dt">txid =</span> txid, <span class="dt">ncbi_dr =</span> ncbi_dr, <span class="dt">v =</span> v, <span class="dt">btchsz =</span> btchsz,
      <span class="dt">mnsql =</span> mnsql, <span class="dt">mxsql =</span> mxsql)
<span class="kw"><a href="http://www.rdocumentation.org/packages/phylotaR/topics/run">run</a></span>(<span class="dt">wd =</span> wd)</code></pre></div>
<p>Compared to running without a restez database, the above code is <a href="https://github.com/AntonelliLab/restez/blob/master/other/phylotar_demo.R"><strong>35%</strong> faster during the download stage</a>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#intoduction">Intoduction</a></li>
      <li><a href="#restez-set-up">Restez set-up</a></li>
      <li><a href="#phylotar-run">phylotaR run</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
