<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Create and Query a Local Copy of GenBank in R • restez</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Create and Query a Local Copy of GenBank in R">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/phylotar.html">phylotaR</a>
    </li>
    <li>
      <a href="../articles/search_and_fetch.html">Search and Fetch</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AntonelliLab/restez#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Create and Query a Local Copy of GenBank in R</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AntonelliLab/restez#readme/blob/master/vignettes/restez.Rmd"><code>vignettes/restez.Rmd</code></a></small>
      <div class="hidden name"><code>restez.Rmd</code></div>

    </div>

    
    
<div id="contents" class="section level1">
<h1 class="hasAnchor">
<a href="#contents" class="anchor"></a>0. Contents</h1>
<ol style="list-style-type: decimal">
<li><strong><a href="#download-genbank">Download GenBank</a></strong></li>
<li><strong><a href="#query">Query your local database</a></strong></li>
<li><strong><a href="#entrez">Use Entrez tools</a></strong></li>
</ol>
</div>
<div id="download-genbank" class="section level1">
<h1 class="hasAnchor">
<a href="#download-genbank" class="anchor"></a>1. Download GenBank</h1>
<p>In the first section of this introduction to <code>restez</code> we will explain how to download parts of GenBank on to your computer and from these downloads, construct a local database. GenBank can be downloaded from NCBI using the File Transfer Protocol (FTP), the online portal for these downloads can be viewed from your web browser, <a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/">FTP NCBI GenBank</a>. GenBank is hosted at this portal as 1000s of compressed sequence files called flatfiles. New versions of these flatfiles are released once a month as new sequence information is uploaded. For information on the latest GenBank release, see the <a href="https://www.ncbi.nlm.nih.gov/genbank/statistics/">NCBI statistics page</a>.</p>
<p><code>restez</code> downloads a selection of these flatfiles, decompresses them and unpacks them into an SQLite database that can be searched using its functions. Because there are potentially huge amounts of sequence information, all of which is unlikely to be of interest to a user, <code>restez</code> allows you to select the sets of flatfiles to download. For the most part these sets of flatfile types are determined by taxonomy: a user can download all primates or plant sequences, for example.</p>
<p>Here, we will explain how to set up <code>restez</code>, download genbank and then create a database from the downloaded files.</p>
<div id="setting-up" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-up" class="anchor"></a>1.1 Setting up</h2>
<p>Before we can download, we need to tell <code>restez</code> where we would like to store the downloaded files and the eventual GenBank database. To do this we use the <code>restez_path_set</code> function. A user provides a system file path to a folder, and within that folder a new folder called ‘restez’ is created. In this example, I will create a <code>restez</code> folder in my desktop.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="co"># On my linux computer and I specify my desktop</span>
fp &lt;-<span class="st"> '~/Desktop'</span>
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(<span class="dt">filepath =</span> fp)</code></pre></div>
<pre><code>... Creating '~/Desktop/restez'
... Creating '~/Desktop/restez/downloads'</code></pre>
<p>Messages are printed to console telling us that the path was created and that a downloads subfolder was created.</p>
<blockquote>
<p><strong>Note</strong>, whenever you intend to use <code>restez</code> you will need to specify the <code>restez</code> path. If you are using it regularly, it might be a good idea to set it using your .Rprofile.</p>
</blockquote>
</div>
<div id="download" class="section level2">
<h2 class="hasAnchor">
<a href="#download" class="anchor"></a>1.2 Download</h2>
<p>Next, we download genbank using <code>gb_download()</code>. This is an interactive function, it looks up the latest GenBank release, parses the release notes and prints to console the available sets of flatfiles that can be downloaded.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gb_download</span>()</code></pre></div>
<pre><code>──────────────────────────────────────────────────────────────────────────────────────────────────────
Looking up latest GenBank release ...
... found release 224
Found 3057 sequence files
──────────────────────────────────────────────────────────────────────────────────────────────────────
Which sequence file types would you like to download?
Choose from those listed below:
● 1  -  'EST (expressed sequence tag)' (485 files available)
● 2  -  'Bacterial' (450 files available)
● 3  -  'Constructed' (362 files available)
● 4  -  'Patent' (323 files available)
● 5  -  'GSS (genome survey sequence)' (304 files available)
● 6  -  'TSA (transcriptome shotgun assembly)' (234 files available)
● 7  -  'Plant sequence entries (including fungi and algae),' (168 files available)
● 8  -  'Invertebrate' (161 files available)
● 9  -  'HTGS (high throughput genomic sequencing)' (154 files available)
● 10  -  'Environmental sampling' (101 files available)
● 11  -  'Other vertebrate' (85 files available)
● 12  -  'Primate' (58 files available)
● 13  -  'Viral' (53 files available)
● 14  -  'Other mammalian' (39 files available)
● 15  -  'Rodent' (30 files available)
● 16  -  'STS (sequence tagged site)' (20 files available)
● 17  -  'HTC (high throughput cDNA sequencing)' (15 files available)
● 18  -  'Synthetic and chimeric' (10 files available)
● 19  -  'Phage' (4 files available)
● 20  -  'Unannotated' (1 files available)
Provide one or more numbers separated by spaces.
e.g. to download all Mammal sequences type: "12 14 15" followed by Enter
Which files would you like to download?
(Press Esc to quit) </code></pre>
<p>We are prompted to provide a number indicating which set we would like to download. In the examples of these tutorials we make use of one of the smaller sets, the rodents, which is number 15 in this example.</p>
<pre><code>(Press Esc to quit) 15
You've selected a total of 30 file types. These represent:
● 'Rodent'
Each file contains about 250 MB of decompressed data.
30 files amounts to about 7.5 GB
Additionally, the resulting SQL database takes about 500 MB per file
In total, the uncompressed files and SQL database should amount to 22.5 GB Is that OK?
Enter any key to continue or press Esc to quit 
──────────────────────────────────────────────────────────────────────────────────────────────────────
Downloading ...
... 'gbrod1.seq' (1 / 30)</code></pre>
<p>Now the download has begun, we just need to wait for it to finish. This can take quite a while.</p>
</div>
<div id="create-a-database" class="section level2">
<h2 class="hasAnchor">
<a href="#create-a-database" class="anchor"></a>1.3 Create a database</h2>
<p>After the download has completed, we need to use <code><a href="../reference/db_create.html">db_create()</a></code> to create our local database. This looks in the downloads folder of our <code>restez</code> path, decompresses any .seq.tar files, breaks these files up into separate GenBank records and adds them to an SQLite database.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/db_create.html">db_create</a></span>()</code></pre></div>
<pre><code>Decompressing and adding 30 files to database ...
... 'gbrod10.seq.gz'(1 / 30)
... ... decompressing
... ... adding
... 'gbrod11.seq.gz'(2 / 30)
... ... decompressing
... ... adding</code></pre>
<p>Again, this can take quite a while.</p>
</div>
<div id="checking-the-setup" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-the-setup" class="anchor"></a>1.4 Checking the setup</h2>
<p>After the download and the database creation steps are complete, we can confirm our setup with the following commands.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="kw"><a href="../reference/restez_path_get.html">restez_path_get</a></span>())
<span class="kw">status_check</span>()</code></pre></div>
<pre><code>[1] "~/Desktop/restez"
Checking setup status ...
... found 31 files in 'downloads/'
... totalling 7.28 GB
... found 'sql_db' of 12.23 GB</code></pre>
<p>The above messages tell us that the <code>restez</code> path is set to the desktop and that the <code>restez</code> folder contains a database of 12.23 GB and that the downloads folder contains all of the rodent downloads, 7.28 GB.</p>
<p>If your status shows the same, then you are ready to move to the next section.</p>
</div>
</div>
<div id="query" class="section level1">
<h1 class="hasAnchor">
<a href="#query" class="anchor"></a>2. Query</h1>
<p>In this section we will explain how you make queries to your local copy of GenBank.</p>
<div id="get-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#get-methods" class="anchor"></a>2.1 Get methods</h2>
</div>
<div id="other-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#other-methods" class="anchor"></a>2.2 Other methods</h2>
</div>
</div>
<div id="entrez" class="section level1">
<h1 class="hasAnchor">
<a href="#entrez" class="anchor"></a>3. Entrez</h1>
<p>It this tutorial, we will explain how entrez wrappers are part of the <code>restez</code> pacakge. These allow a user to make use of the local GenBank using functions that were built for <a href="https://github.com/ropensci/rentrez"><code>rentrez</code></a>. This minimises the amount of coding changes required for any Entrez dependent code.</p>
<div id="rentrez-wrappers" class="section level2">
<h2 class="hasAnchor">
<a href="#rentrez-wrappers" class="anchor"></a>3.1 rentrez wrappers</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#contents">0. Contents</a></li>
      <li>
<a href="#download-genbank">1. Download GenBank</a><ul class="nav nav-pills nav-stacked">
<li><a href="#setting-up">1.1 Setting up</a></li>
      <li><a href="#download">1.2 Download</a></li>
      <li><a href="#create-a-database">1.3 Create a database</a></li>
      <li><a href="#checking-the-setup">1.4 Checking the setup</a></li>
      </ul>
</li>
      <li>
<a href="#query">2. Query</a><ul class="nav nav-pills nav-stacked">
<li><a href="#get-methods">2.1 Get methods</a></li>
      <li><a href="#other-methods">2.2 Other methods</a></li>
      </ul>
</li>
      <li>
<a href="#entrez">3. Entrez</a><ul class="nav nav-pills nav-stacked">
<li><a href="#rentrez-wrappers">3.1 rentrez wrappers</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
