<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Create and Query a Local Copy of GenBank in R • restez</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Create and Query a Local Copy of GenBank in R">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_rodents.html">1. Build a database of all rodents</a>
    </li>
    <li>
      <a href="../articles/2_search_and_fetch.html">2. How to search for and fetch sequences</a>
    </li>
    <li>
      <a href="../articles/3_parsing.html">3. Advanced parsing of a GenBank record</a>
    </li>
    <li>
      <a href="../articles/4_phylotar.html">4. Running phylotaR with restez</a>
    </li>
    <li>
      <a href="../articles/5_tips_and_tricks.html">5. Tips and Tricks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/restez#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Create and Query a Local Copy of GenBank in R</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/restez#readme/blob/master/vignettes/restez.Rmd"><code>vignettes/restez.Rmd</code></a></small>
      <div class="hidden name"><code>restez.Rmd</code></div>

    </div>

    
    
<div id="download-genbank" class="section level1">
<h1 class="hasAnchor">
<a href="#download-genbank" class="anchor"></a>1. Download GenBank</h1>
<p>In the first section of this introduction to <code>restez</code> we will explain how to download parts of GenBank on to your computer and from these downloads, construct a local database. GenBank can be downloaded from NCBI using the File Transfer Protocol (FTP), the online portal for these downloads can be viewed from your web browser, <a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/">FTP NCBI GenBank</a>. GenBank is hosted at this portal as 1000s of compressed sequence files called flatfiles. New versions of these flatfiles are released once a month as new sequence information is uploaded. For information on the latest GenBank release, see the <a href="https://www.ncbi.nlm.nih.gov/genbank/statistics/">NCBI statistics page</a>.</p>
<p><code>restez</code> downloads a selection of these flatfiles and unpacks them into an SQL-like database (specifically, <a href="https://www.monetdb.org/">MonetDB</a>) that can be queried using <code>restez</code>’s functions. Because there are potentially huge amounts of sequence information, all of which is unlikely to be of interest to a user, <code>restez</code> allows you to select the sets of flatfiles to download based on GenBank’s domains. For the most part these sets of flatfile types are determined by taxonomy, for example, a user can download all primate or plant sequences.</p>
<p>Here, we will explain how to set up <code>restez</code>, download genbank and then create a database from the downloaded files.</p>
<div id="setting-up" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-up" class="anchor"></a>1.1 Setting up</h2>
<p>Before we can download anything, we need to tell <code>restez</code> where we would like to store the downloaded files and the eventual GenBank database. To do this we use the <code>restez_path_set</code> function. With this function a user can provide a system file path to a folder within which the function will create a new folder called ‘restez’. In this example, we will create a <code>restez</code> folder in a temporary folder.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; restez v1.0.0</span>
<span class="co">#&gt; -------------</span>
<span class="co">#&gt; Remember to restez_path_set() and, then, restez_connect()</span>
<span class="co"># rstz_pth &lt;- tempdir()</span>
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(<span class="dt">filepath =</span> rstz_pth)
<span class="co">#&gt; ... Creating '/var/folders/ps/g89999v12490dmp0jnsfmykm0043m3/T//RtmpWv9WyX/restez'</span>
<span class="co">#&gt; ... Creating '/var/folders/ps/g89999v12490dmp0jnsfmykm0043m3/T//RtmpWv9WyX/restez/downloads'</span></code></pre></div>
<p>A user must always set the <code>restez</code> path in order to use the <code>restez</code> library and we are reminded to do this whenever we load the package. (We are also reminded to run <code>restez_connect</code> but we’ll come to that a bit later.) Additionally, by running <code>restez_path_set</code> on a new folder messages are printed to console telling us that the path was created and that a downloads subfolder was created.</p>
<blockquote>
<p><strong>Note</strong>, whenever you intend to use <code>restez</code> you will need to specify the <code>restez</code> path. If you are using it regularly, it might be a good idea to set it using your .Rprofile.</p>
</blockquote>
</div>
<div id="download" class="section level2">
<h2 class="hasAnchor">
<a href="#download" class="anchor"></a>1.2 Download</h2>
<p>Now we can download GenBank files using <code><a href="../reference/db_download.html">db_download()</a></code>. This is an interactive function, it looks up the latest GenBank release, parses the release notes and prints to console the available sets of flatfiles that can be downloaded. For this example, we will select the smallest available domain which is ‘Unannotated’ and can be pre-specified with ‘20’. Normally, you would run this function without the <code>preselection</code> argument and just wait for the function to prompt you for a domain selection.</p>
<p>After launching, the download function will ask you whether it is OK to download and set up a database for the selected sequence files. If the selection is likely to be large you can always quit the process using either <code>Esc</code> or <code>Ctrl+c</code>. Please note that the diskspace estimates are preliminary. The package trys to be conservative in its estimates.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/db_download.html">db_download</a></span>(<span class="dt">preselection =</span> <span class="st">'20'</span>)
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Looking up latest GenBank release ...</span>
<span class="co">#&gt; ... release number 228</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; ... found 3232 sequence files</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Which sequence file types would you like to download?</span>
<span class="co">#&gt; Choose from those listed below:</span>
<span class="co">#&gt; ● 1  - 'Bacterial'</span>
<span class="co">#&gt;         542 files and 123 GB</span>
<span class="co">#&gt; ● 2  - 'EST (expressed sequence tag)'</span>
<span class="co">#&gt;         486 files and 243 GB</span>
<span class="co">#&gt; ● 3  - 'Constructed'</span>
<span class="co">#&gt;         372 files and 84.6 GB</span>
<span class="co">#&gt; ● 4  - 'Patent'</span>
<span class="co">#&gt;         340 files and 77.9 GB</span>
<span class="co">#&gt; ● 5  - 'GSS (genome survey sequence)'</span>
<span class="co">#&gt;         308 files and 117 GB</span>
<span class="co">#&gt; ● 6  - 'TSA (transcriptome shotgun assembly)'</span>
<span class="co">#&gt;         234 files and 53.9 GB</span>
<span class="co">#&gt; ● 7  - 'Plant sequence entries (including fungi and algae)'</span>
<span class="co">#&gt;         232 files and 62.8 GB</span>
<span class="co">#&gt; ● 8  - 'HTGS (high throughput genomic sequencing)'</span>
<span class="co">#&gt;         155 files and 36.7 GB</span>
<span class="co">#&gt; ● 9  - 'Invertebrate'</span>
<span class="co">#&gt;         112 files and 24.8 GB</span>
<span class="co">#&gt; ● 10 - 'Environmental sampling'</span>
<span class="co">#&gt;         103 files and 24.1 GB</span>
<span class="co">#&gt; ● 11 - 'Other vertebrate'</span>
<span class="co">#&gt;         94 files and 19.4 GB</span>
<span class="co">#&gt; ● 12 - 'Primate'</span>
<span class="co">#&gt;         59 files and 13.7 GB</span>
<span class="co">#&gt; ● 13 - 'Viral'</span>
<span class="co">#&gt;         58 files and 13.1 GB</span>
<span class="co">#&gt; ● 14 - 'Other mammalian'</span>
<span class="co">#&gt;         55 files and 9.44 GB</span>
<span class="co">#&gt; ● 15 - 'Rodent'</span>
<span class="co">#&gt;         31 files and 7.32 GB</span>
<span class="co">#&gt; ● 16 - 'STS (sequence tagged site)'</span>
<span class="co">#&gt;         20 files and 4.45 GB</span>
<span class="co">#&gt; ● 17 - 'HTC (high throughput cDNA sequencing)'</span>
<span class="co">#&gt;         15 files and 3.43 GB</span>
<span class="co">#&gt; ● 18 - 'Synthetic and chimeric'</span>
<span class="co">#&gt;         10 files and 2.42 GB</span>
<span class="co">#&gt; ● 19 - 'Phage'</span>
<span class="co">#&gt;         5 files and 1.12 GB</span>
<span class="co">#&gt; ● 20 - 'Unannotated'</span>
<span class="co">#&gt;         1 files and 0.00111 GB</span>
<span class="co">#&gt; Provide one or more numbers separated by spaces.</span>
<span class="co">#&gt; e.g. to download all Mammalian sequences, type: "12 14 15" followed by Enter</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Which files would you like to download?</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; You've selected a total of 1 file(s) and 0.00111 GB of uncompressed data.</span>
<span class="co">#&gt; These represent: </span>
<span class="co">#&gt; ● 'Unannotated'</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Based on stated GenBank files sizes, we estimate ... </span>
<span class="co">#&gt; ... 0.000222 GB for  compressed, downloaded files</span>
<span class="co">#&gt; ... 0.00135 GB for the SQL database</span>
<span class="co">#&gt; Leading to a total of 0.00158 GB</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Please note, the real sizes of the database and its downloads cannot be</span>
<span class="co">#&gt; accurately predicted beforehand.</span>
<span class="co">#&gt; These are just estimates, actual sizes may differ by up to a factor of two.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Is this OK?</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Downloading ...</span>
<span class="co">#&gt; ... 'gbuna1.seq' (1/1)</span>
<span class="co">#&gt; Done. Enjoy your day.</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Now the download has begun, we just need to wait for it to finish. For the largest domains, this can take quite a while and may be prone to server errors. If you ever encounter a failed downloaded file it is possible to re-run the <code>db_download</code> function again and – so long as the same selection is made – the function will only download the missing files.</p>
</div>
<div id="create-a-database" class="section level2">
<h2 class="hasAnchor">
<a href="#create-a-database" class="anchor"></a>1.3 Create a database</h2>
<p>After the download has completed, we need to use <code><a href="../reference/db_create.html">db_create()</a></code> to create our local database. This looks in the downloads folder of our <code>restez</code> path, breaks these files up into separate GenBank records and adds them to the SQL-like database. Again, for very large collections of sequences this can take quite a while.</p>
<p>Before running the <code>db_create</code> function, however, we must first establish a connection to the database in the <code>restez</code> path using the <code>restez_connect</code> function. As with SQL databases we must always make sure to connect and then disconnect after we have interacted with a database.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
<span class="kw"><a href="../reference/db_create.html">db_create</a></span>()
<span class="co">#&gt; Adding 1 file(s) to the database ...</span>
<span class="co">#&gt; ... [32m'gbuna1.seq.gz'[39m([34m1 / 1[39m)</span>
<span class="co">#&gt; Done.</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
<p><code><a href="../reference/db_create.html">db_create()</a></code> allows a user to specify minimum and maximum sequence lengths. It’s always a good idea to limit the number of sequences in a database so that look-up times are faster. If you know you are only interested in certain lengths of sequences it is a good idea to limit the sequences in the database at this stage. You can always run <code><a href="../reference/db_create.html">db_create()</a></code> again to change the limits. You will simply need to delete the original database first with <code>db_delete</code>.</p>
</div>
<div id="checking-the-setup" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-the-setup" class="anchor"></a>1.4 Checking the setup</h2>
<p>After the download and the database creation steps are complete, we can confirm our setup using <code>restez_status</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rstz_pth)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
<span class="kw"><a href="../reference/restez_status.html">restez_status</a></span>()
<span class="co">#&gt; Checking setup status at  ...</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Restez path ...</span>
<span class="co">#&gt; ... Path '/var/folders/ps/g89999v12490dmp0jnsfmykm0043m3/T//RtmpWv9WyX/restez'</span>
<span class="co">#&gt; ... Does path exist? 'Yes'</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Download ...</span>
<span class="co">#&gt; ... Path '/var/folders/ps/g89999v12490dmp0jnsfmykm0043m3/T//RtmpWv9WyX/restez/downloads'</span>
<span class="co">#&gt; ... Does path exist? 'Yes'</span>
<span class="co">#&gt; ... N. files 2</span>
<span class="co">#&gt; ... N. GBs 0</span>
<span class="co">#&gt; ... GenBank division selections 'Unannotated'</span>
<span class="co">#&gt; ... GenBank Release 228</span>
<span class="co">#&gt; ... Last updated '2018-11-15 14:05:40'</span>
<span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; Database ...</span>
<span class="co">#&gt; ... Path '/var/folders/ps/g89999v12490dmp0jnsfmykm0043m3/T//RtmpWv9WyX/restez/sql_db'</span>
<span class="co">#&gt; ... Does path exist? 'Yes'</span>
<span class="co">#&gt; ... N. GBs 0</span>
<span class="co">#&gt; ... Is database connected? 'Yes'</span>
<span class="co">#&gt; ... Does the database have data? 'Yes'</span>
<span class="co">#&gt; ... Number of sequences 448</span>
<span class="co">#&gt; ... Min. sequence length 0</span>
<span class="co">#&gt; ... Max. sequence length Inf</span>
<span class="co">#&gt; ... Last_updated '2018-11-15 14:05:44'</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
<p>The status function allows a user to always touch base with their <code>restez</code> set-up. It can be run at any point, before and/or after download or database creation. It is also designed to provide useful messages to a user for what they need to do next in order for them to make queries to their database. If you ever get confused, run <code>restez_status</code> to see what is happening.</p>
<p>Additionally, if you are developing your own functions that depend on <code>restez</code>, you can use <code>restez_ready</code> which will simply return TRUE or FALSE on whether the database can be queried.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rstz_pth)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
<span class="cf">if</span> (<span class="kw"><a href="../reference/restez_ready.html">restez_ready</a></span>()) {
  <span class="kw">print</span>(<span class="st">'Database is ready to be queried!'</span>)
} <span class="cf">else</span> {
  <span class="kw">print</span>(<span class="st">'Database cannot be queried :-('</span>)
}
<span class="co">#&gt; [1] "Database is ready to be queried!"</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
</div>
</div>
<div id="query" class="section level1">
<h1 class="hasAnchor">
<a href="#query" class="anchor"></a>2. Query</h1>
<p>Once a restez database has been set up we can query the database using the <code>gb_*_get()</code> functions. These functions allow us to retrieve specific columns in the SQL-like database: ‘sequence’, ‘definition’, ‘accession’, ‘version’ and ‘organism’. Also, they allow us to get the whole text formatted record and sequence data in fasta format. In this example, we can use <code><a href="../reference/list_db_ids.html">list_db_ids()</a></code> to identify accession numbers in the database. We could also use <code>entrez_search()</code> provided the database contains sequences of interest to us, see <a href="https://ropensci.github.io/restez/articles/2_search_and_fetch.html">‘search and fetch’</a>.</p>
<blockquote>
<p>Always remember to disconnect after you have finished your querying!</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rstz_pth)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
ids &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw"><a href="../reference/list_db_ids.html">list_db_ids</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">n =</span> <span class="dv">100</span>))
(id &lt;-<span class="st"> </span><span class="kw">sample</span>(ids, <span class="dv">1</span>))
<span class="co">#&gt; [1] "AF040893"</span>
<span class="co"># sequence</span>
<span class="kw">str</span>(<span class="kw"><a href="../reference/gb_sequence_get.html">gb_sequence_get</a></span>(id))
<span class="co">#&gt;  Named chr "CCCTCGAAGCCCTCATCGCTGAGGCTGCCCCGGATACCGAGGTTCTTTACAAAGACCGGATGGACTCCAGCAACGACACCTCGCTGGTAACAGCCCTCATGACGACCTATGACGGCACTCTTGAGG"</span>
<span class="co">#&gt;  - attr(*, "names")= chr "AF040893"</span>
<span class="co"># definition</span>
(<span class="kw"><a href="../reference/gb_definition_get.html">gb_definition_get</a></span>(id))
<span class="co">#&gt;                        AF040893 </span>
<span class="co">#&gt; "Unidentified RNA clone P10.11"</span>
<span class="co"># version</span>
(<span class="kw"><a href="../reference/gb_version_get.html">gb_version_get</a></span>(id))
<span class="co">#&gt;     AF040893 </span>
<span class="co">#&gt; "AF040893.1"</span>
<span class="co"># organism</span>
(<span class="kw"><a href="../reference/gb_organism_get.html">gb_organism_get</a></span>(id))
<span class="co">#&gt;       AF040893 </span>
<span class="co">#&gt; "unidentified"</span>
<span class="co"># fasta</span>
<span class="kw">cat</span>(<span class="kw"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span>(id))
<span class="co">#&gt; &gt;AF040893.1 Unidentified RNA clone P10.11</span>
<span class="co">#&gt; CCCTCGAAGCCCTCATCGCTGAGGCTGCCCCGGATACCGAGGTTCTTTACAAAGACCGGATGGACTCCAG</span>
<span class="co">#&gt; CAACGACACCTCGCTGGTAACAGCCCTCATGACGACCTATGACGGCACTCTTGAGG</span>
<span class="co"># Note, for large databases these requests can take a long time to complete.</span>
<span class="co"># Always try and limit the size of the database by specifying min and max</span>
<span class="co"># sequence lengths with db_create()</span>
<span class="co"># Also note, if an id is not present in the database nothing is returned</span>
<span class="kw">cat</span>(<span class="kw"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span>(<span class="dt">id =</span> <span class="kw">c</span>(id, <span class="st">'notanid'</span>)))
<span class="co">#&gt; &gt;AF040893.1 Unidentified RNA clone P10.11</span>
<span class="co">#&gt; CCCTCGAAGCCCTCATCGCTGAGGCTGCCCCGGATACCGAGGTTCTTTACAAAGACCGGATGGACTCCAG</span>
<span class="co">#&gt; CAACGACACCTCGCTGGTAACAGCCCTCATGACGACCTATGACGGCACTCTTGAGG</span>
<span class="co"># Always remember to disconnect</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
<p>Additionally, for more flexibility and options for extracting sequence record information see <a href="https://ropensci.github.io/restez/articles/3_parsing.html">GenBank record parsing</a>.</p>
</div>
<div id="entrez" class="section level1">
<h1 class="hasAnchor">
<a href="#entrez" class="anchor"></a>3. Entrez</h1>
<p>Entrez wrappers are part of the <code>restez</code> package. These allow a user to make use of the local GenBank using functions that were built for <a href="https://github.com/ropensci/rentrez"><code>rentrez</code></a>. This minimises the amount of coding changes required for any Entrez dependent code.</p>
<blockquote>
<p>Currently, only <code><a href="../reference/entrez_fetch.html">entrez_fetch()</a></code> is available with restez and only text formatted rettypes are allowed.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(rstz_pth)
<span class="kw"><a href="../reference/restez_connect.html">restez_connect</a></span>()
<span class="co">#&gt; Remember to run `restez_disconnect()`</span>
ids &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw"><a href="../reference/list_db_ids.html">list_db_ids</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">n =</span> <span class="dv">100</span>))
(id &lt;-<span class="st"> </span><span class="kw">sample</span>(ids, <span class="dv">1</span>))
<span class="co">#&gt; [1] "AF298107"</span>
<span class="co"># get fasta sequences with entrez</span>
res &lt;-<span class="st"> </span>restez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/restez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">id =</span> id, <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
<span class="kw">cat</span>(res)
<span class="co">#&gt; &gt;AF298107.1 Unidentified clone B33 DNA sequence from ocean beach sand</span>
<span class="co">#&gt; GATCCTCGAACTACGGCGCCGCGGCGACCCGGGCCCAAGATCGCCTCAACGAGGCGCTCCGGCCCCACGG</span>
<span class="co">#&gt; CGTCGTCGTGACCCAGATGGTCACGCCGCGACCCAAATTCAACGCCGACTACGAGCACCTGATCGAGGAG</span>
<span class="co">#&gt; CGCAACCGCCTCCGCAACGAGCTCGAGGTCATCAAGTCCAACCTCGAGCGGGCCGAGACCGACCGCCAGC</span>
<span class="co">#&gt; GTACGCTGTCCGAGGTCGACCGCGACGAGAACAAGAAGATCCAGGAGCGCCGCGCCGAGCTCGAGTCTGA</span>
<span class="co">#&gt; GCTCGCGACCGCGGTCGCAGATCGCGAGAACGCCATCCGCGCGGCCGACACCTACCGAATCGAGACGATC</span>
<span class="co">#&gt; GGCGTCGGCCAGGCCAAGCTCGCCGCGGCCCGGCGCAAGGCCGAGGAGCTCCAGGGCCAGCTCGAGGCCG</span>
<span class="co">#&gt; ACTACCGGGCCAAGAAGGCCGAGGTCGAGGCGTTTCGGACCCAACCCGTCGAGCGCGTGATGGAGCGCCT</span>
<span class="co">#&gt; CGGCGAGAGCCTCGAGGACGTGACGATCAACATTGAGCCGTGGGCCAACGACGCCNACCCTTACGCGTGG</span>
<span class="co">#&gt; TCTNGACCAATAGGAGACGGACCATGGAAAACTCACCACTGCTGCAATGGTTCTGATCGCTTCTTCCNAT</span>
<span class="co">#&gt; CTTGGNCTTCCCGGATGATGGTCACAACTCNACTCGGC</span>
<span class="co"># entrez_fetch will also search via Entrez for any ids not in the db</span>
plant_sequence &lt;-<span class="st"> 'AY952423'</span>
res &lt;-<span class="st"> </span>restez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/restez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">id =</span> <span class="kw">c</span>(id, plant_sequence),
                            <span class="dt">rettype =</span> <span class="st">'fasta'</span>)
<span class="co">#&gt; [1] id(s) are unavailable locally, searching online.</span>
<span class="kw">cat</span>(res)
<span class="co">#&gt; &gt;AF298107.1 Unidentified clone B33 DNA sequence from ocean beach sand</span>
<span class="co">#&gt; GATCCTCGAACTACGGCGCCGCGGCGACCCGGGCCCAAGATCGCCTCAACGAGGCGCTCCGGCCCCACGG</span>
<span class="co">#&gt; CGTCGTCGTGACCCAGATGGTCACGCCGCGACCCAAATTCAACGCCGACTACGAGCACCTGATCGAGGAG</span>
<span class="co">#&gt; CGCAACCGCCTCCGCAACGAGCTCGAGGTCATCAAGTCCAACCTCGAGCGGGCCGAGACCGACCGCCAGC</span>
<span class="co">#&gt; GTACGCTGTCCGAGGTCGACCGCGACGAGAACAAGAAGATCCAGGAGCGCCGCGCCGAGCTCGAGTCTGA</span>
<span class="co">#&gt; GCTCGCGACCGCGGTCGCAGATCGCGAGAACGCCATCCGCGCGGCCGACACCTACCGAATCGAGACGATC</span>
<span class="co">#&gt; GGCGTCGGCCAGGCCAAGCTCGCCGCGGCCCGGCGCAAGGCCGAGGAGCTCCAGGGCCAGCTCGAGGCCG</span>
<span class="co">#&gt; ACTACCGGGCCAAGAAGGCCGAGGTCGAGGCGTTTCGGACCCAACCCGTCGAGCGCGTGATGGAGCGCCT</span>
<span class="co">#&gt; CGGCGAGAGCCTCGAGGACGTGACGATCAACATTGAGCCGTGGGCCAACGACGCCNACCCTTACGCGTGG</span>
<span class="co">#&gt; TCTNGACCAATAGGAGACGGACCATGGAAAACTCACCACTGCTGCAATGGTTCTGATCGCTTCTTCCNAT</span>
<span class="co">#&gt; CTTGGNCTTCCCGGATGATGGTCACAACTCNACTCGGC</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; &gt;AY952423.1 Livistona chinensis tRNA-Lys (trnK) gene, partial sequence; and matK gene, complete sequence; chloroplast</span>
<span class="co">#&gt; ATTGGGGTTGCTAACTCAACGGTAGAGTACTCGGCTTTTAAGTGCGACTATCATCTTTTACACATTTGGA</span>
<span class="co">#&gt; TGAAGTAAGGAATTCGTCCAGACTATTGGTAGAGTCTATAAGACCACGACTGATCCTGAAAGGTAATGAA</span>
<span class="co">#&gt; TGGAAAAAATAGCATGTCGTACGTAATACAATGAGAAACTTGTAATTTCTTATTGTAATTTTTTAAGTAG</span>
<span class="co">#&gt; AACTTTGAGTTTATCCTTACTGGATCATTACAAAAATATTGTATTTTATTTTTGGAAGGGGACGAAAAAA</span>
<span class="co">#&gt; AGGAAATTCCCAACATTTATTGTTTGGTCTAATGAATAAATGGATAGGGGCCTAGGGTAGGGCCCAATTT</span>
<span class="co">#&gt; TTGTAAAACAAAAAGCAACGAGCTTATGTTCTTAATTTGAATAATTACCCGATCTAATTAGATGTTAAAA</span>
<span class="co">#&gt; ATAAATTAGTGCCAGATGTGGTAAAGGGTTCTACTGTAAGTGGACCTTTTTTTTTTTTTTTTATGAATCC</span>
<span class="co">#&gt; TACCTATTATCTATTATGGATTAAAGATGGATGTGTATAAGAAGAAGTATACTGATAAAGAGAATTTTTC</span>
<span class="co">#&gt; CAAAGTCAAAAGAGCAATCGGGTTGCAAAAATAAAGGATTTTTACCTCCGAGAATTATAAATTAATTGGA</span>
<span class="co">#&gt; TCAAAAGGAGAGGAAAAAGTCTGTGATTGGACTCCTTCTATCCGCGGGTATGGGTATATAGTAGGTATAT</span>
<span class="co">#&gt; ATGTATATTTGTATACTATATAAATTACATGCCCTGTTCTGACCGTATTGCACTATGTATTATTTGATAA</span>
<span class="co">#&gt; TCCAAGAAATGCCTCCTACTTCTGGTTCAAGTAGAAATGAAAATAGAAGAATTACAAGAATATTTAGAAA</span>
<span class="co">#&gt; AAGATAGATCTCGGCAACAACACTTCCTATACCCACTTTTCTTTCAGGAGTATATTTATGCACTTGCTCA</span>
<span class="co">#&gt; TGATTATGGGTTTAAAGGGTTCGATTTTTTACGAACCTATGGAAATTGGGGGTTATGATAATAAATCTAG</span>
<span class="co">#&gt; TTCAGTACTTGTAAAACATTTAATTACTCGAATGTATCAACAGAATTATTTGATTTATTCTGTTAATGAA</span>
<span class="co">#&gt; TCTAACCAAAATCGATTGATTGAGCATAACAATTCTTTTTATTCTCAAATGATATCTGAAGTTTTTGCGA</span>
<span class="co">#&gt; TCATTGCAGAAATTCCATTCTCTCAGCAATTACTATTTTCTCTTCGAGGAAAAAAGAATACCAAAATCTC</span>
<span class="co">#&gt; AGACTTTACGATCTATTCATTCAATATTTCCCTTTTTAGAAGACAAATTATCACATTTAAACTATGTGTC</span>
<span class="co">#&gt; AGATATATTAATACCCTATCCCATCCATTTGGAAATCTTGGTGCAAATTCTTCAATGCTGGATCCAAGAT</span>
<span class="co">#&gt; GTTTCTTCTTTGCATTTATTGCGATTCTTTCTCCACGAACATCATAATGGGAATAGTTTTTTTTTTCCAA</span>
<span class="co">#&gt; AGAAATCCTTTTCAAAAGAAAATAAAAGACTCTTTCGATTCCTATATAATTCTTATGTATCTGAATGTGA</span>
<span class="co">#&gt; ATTTGTCTTAGTGTTTCTTCGTAAACAATCCTCTTATTTACAATCAAAATCCTATGGAATCTTTCTTGAG</span>
<span class="co">#&gt; CGAACACATTTCTATGGAAGAATGGAACATCTTATAGTAGTGTGTCATAATTATTGTCAGAAGGCCTTTT</span>
<span class="co">#&gt; GGGTCTTCAAGGATCCTTTTATGCATTATGTTCGATATCAAGGAAAAGCAATTCTGGCATCAAAAGGATC</span>
<span class="co">#&gt; TTATCTTTTGATGAAGAAATGGAGATGTCATCTTGTCAATTTCTGGCAATATTATTTTCATTTTTGGGCT</span>
<span class="co">#&gt; CAGCCTTACAGAATTTCAATAAACCAATTAGGAAATCATTCCTTCTATTTTCTCGGTTATCTTTCAAGTG</span>
<span class="co">#&gt; TATTAAAAAATACTTCGTCTGTAAGGAATCAAATGCTAGAGAATTCCTTTTTAATAGATACTATTACTAA</span>
<span class="co">#&gt; TAAATTGGATACCATAGTCCCAGTTCTTCCTCTTATTGGATCTTTGTCTAAAGCTAAATTTTGTACCGTA</span>
<span class="co">#&gt; TCCGGGCATCCTAGTAGTAAGCCAATCTGGACGGATTTATCGGATTCTGATATTATTGATAGATTTGGTC</span>
<span class="co">#&gt; GGATATGTAGAAATCTTTCTCATTATTATAGTGGATCCTCAAAAAAACAGAGCTTATATCGAATAAGGTA</span>
<span class="co">#&gt; TATACTTCGACTTTCTTGTGCTAGAACTTTAGCTCGTAAACATAAAAGTACAGTACGTGCTTTTTTGCAA</span>
<span class="co">#&gt; AGATTAGGTTCGGAATTATTAGAAGAATTCTTTACAGAAGAAGAAGGAGTTGTTTTTTTGATTTCCCAAA</span>
<span class="co">#&gt; AGAACAAAACCTCTTTTCCTCTCTATAGGTCACATAGAGAACGCATTTGGTATTTGGATATTATCCATAT</span>
<span class="co">#&gt; TAATGAATTGGTGAATTCATTTATGATGGGGCGATAAGCCCCTATAAAATAAGAAATATAAATTTTTTCT</span>
<span class="co">#&gt; AATGTCTAATAAATAGACGACAAATTCATTAATTTTCATTCTGAAATGCTCATCTAGTAGTGTAGTGATT</span>
<span class="co">#&gt; GAATCAACTGAGTATTCAAAATTTTTAGACAAACTTCTAGGGATAGAAGTTTGTTTTATCTGTATACATA</span>
<span class="co">#&gt; GGTAAAGTCGTGTGCAATGAAAAATGCAAGCACGATTTGGGGAGAGATAATTTTCTCTATTGTAACAAAT</span>
<span class="co">#&gt; AAAAATTATCTACTCCATCCGACTAGTTAATCG</span>
<span class="kw"><a href="../reference/restez_disconnect.html">restez_disconnect</a></span>()</code></pre></div>
</div>
<div id="more-information" class="section level1">
<h1 class="hasAnchor">
<a href="#more-information" class="anchor"></a>4. More information</h1>
<p>For more information about <code>restez</code> see the other tutorials:</p>
<ol style="list-style-type: decimal">
<li><a href="https://ropensci.github.io/restez/articles/1_rodents.html">Build a database of all rodents</a></li>
<li><a href="https://ropensci.github.io/restez/articles/2_search_and_fetch.html">How to search for and fetch sequences</a></li>
<li><a href="https://ropensci.github.io/restez/articles/3_parsing.html">Advanced parsing of a GenBank record</a></li>
<li><a href="https://ropensci.github.io/restez/articles/4_phylotar.html">Running phylotaR with restez</a></li>
<li><a href="https://ropensci.github.io/restez/articles/5_tips_and_tricks.html">Tips and tricks</a></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#download-genbank">1. Download GenBank</a><ul class="nav nav-pills nav-stacked">
<li><a href="#setting-up">1.1 Setting up</a></li>
      <li><a href="#download">1.2 Download</a></li>
      <li><a href="#create-a-database">1.3 Create a database</a></li>
      <li><a href="#checking-the-setup">1.4 Checking the setup</a></li>
      </ul>
</li>
      <li><a href="#query">2. Query</a></li>
      <li><a href="#entrez">3. Entrez</a></li>
      <li><a href="#more-information">4. More information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
