<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Search and Fetch • restez</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Search and Fetch">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">restez</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/restez.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/parsing.html">GenBank record parsing</a>
    </li>
    <li>
      <a href="../articles/phylotar.html">phylotaR</a>
    </li>
    <li>
      <a href="../articles/search_and_fetch.html">Search and Fetch</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AntonelliLab/restez#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Search and Fetch</h1>
            
            <h4 class="date">2018-06-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/AntonelliLab/restez#readme/blob/master/vignettes/search_and_fetch.Rmd"><code>vignettes/search_and_fetch.Rmd</code></a></small>
      <div class="hidden name"><code>search_and_fetch.Rmd</code></div>

    </div>

    
    
<p>In this demonstration we are going to search NCBI with a custom search term, download accession IDs, and retrieve fasta sequences using <code>restez</code>.</p>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<p>First, we need to download and setup the restez database. When we use <code>gb_download()</code> we are going to select ‘rodents’. We are downloading rodents because they are the smallest available set of sequences, minismising the download time for this demonstration. If you have already downloaded the rodents flatfiles as explained in previous tutorials, then simply set your restez path to wherever you previously set your restez path.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(restez)
fp &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">'demo'</span>)
<span class="kw">dir.create</span>(fp)
<span class="kw"><a href="../reference/restez_path_set.html">restez_path_set</a></span>(fp)
<span class="kw">gb_download</span>()
<span class="kw"><a href="../reference/db_create.html">db_create</a></span>()</code></pre></div>
</div>
<div id="search-ncbi" class="section level2">
<h2 class="hasAnchor">
<a href="#search-ncbi" class="anchor"></a>Search NCBI</h2>
<p>After setting up the database, we can then identify the sequences we are interested in. For this, we cannot use restez to perform a search, because the download just doesn’t contain enough metadata for the full range of possible search arguments.</p>
<p>Let’s pretend we’re intersted in COI sequences for a group of Rodents called the Sciuromorpha. We can create an NCBI search term, use <code>entrez_search</code> to perform the search and then retrieve the accession IDs using <code>entrez_fetch</code> and the <code>acc</code> rettype. <code>entrez_fetch</code> returns a single text that will need to be split up by the newline character. Finally, we should drop any version number after the downloaded accessions. This is not strictly necessary, but it makes it easier to check our results from restez later.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># 33553 - Sciuromorpha - squirrel-like things</span>
search_term &lt;-<span class="st"> 'txid33553[Organism:exp] AND COI [GENE]'</span>
search_object &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">term =</span> search_term,
                                        <span class="dt">use_history =</span> <span class="ot">TRUE</span>, <span class="dt">retmax =</span> <span class="dv">0</span>)
accessions &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>,
                                    <span class="dt">web_history =</span> search_object<span class="op">$</span>web_history,
                                    <span class="dt">rettype =</span> <span class="st">'acc'</span>)
accessions &lt;-<span class="st"> </span><span class="kw">strsplit</span>(<span class="dt">x =</span> accessions, <span class="dt">split =</span> <span class="st">'</span><span class="ch">\\</span><span class="st">n'</span>)[[<span class="dv">1</span>]]
accessions &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="dt">pattern =</span> <span class="st">'</span><span class="ch">\\</span><span class="st">.[0-9]+'</span>, <span class="dt">replacement =</span> <span class="st">''</span>, <span class="dt">x =</span> accessions)</code></pre></div>
</div>
<div id="retrieve-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-sequences" class="anchor"></a>Retrieve sequences</h2>
<p>To fetch the sequences from the database we set up, we can just use the <code>gb_fasta_get</code> function. This is a lot faster than using the Entrez API for very large requests, because NCBI limits the number of requests per user, often to as lower as 300 items per request with varying time delays.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">coi_sequences &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gb_fasta_get.html">gb_fasta_get</a></span>(<span class="dt">id =</span> accessions)
<span class="co"># are all accessions in results?</span>
<span class="kw">all</span>(accessions <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(coi_sequences))
<span class="co"># .... no</span></code></pre></div>
<p>When exploring the above results however we notice that not all the accession IDs that were provided are in the returned <code>coi_sequences</code>. This is because restez only downloads GenBank sequences, and these ‘missing’ sequence IDs are RefSeq sequences. Luckily, it seems all RefSeq sequences are also deposited in GenBank under a different ID, so no sequence information is missing from our results. We can look up the alternative ID and test whether it is in our <code>accesssions</code> vector.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># are all accessions in results?</span>
<span class="kw">all</span>(accessions <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(coi_sequences))
<span class="co"># .... no</span>
accessions[<span class="op">!</span>accessions <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(coi_sequences)]
<span class="co"># NC* refers to RefSeq sequences and are not currently available through restez</span>
<span class="co"># The sequence however exists in GB under a different id which we can find like so</span>
smmry &lt;-<span class="st"> </span>rentrez<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/rentrez/topics/entrez_summary">entrez_summary</a></span>(<span class="dt">db =</span> <span class="st">'nucleotide'</span>, <span class="dt">id =</span> <span class="st">'NC_027278'</span>)
<span class="co"># This ID does exist in our results.</span>
smmry<span class="op">$</span>assemblyacc <span class="op">%in%</span><span class="st"> </span>accessions</code></pre></div>
</div>
<div id="take-down" class="section level2">
<h2 class="hasAnchor">
<a href="#take-down" class="anchor"></a>Take down</h2>
<p>Finally, because this is only a demonstration. It might be a good idea to delete the database we built to free up hard disk space.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/db_delete.html">db_delete</a></span>()
<span class="kw"><a href="../reference/restez_path_check.html">restez_path_check</a></span>()  <span class="co"># confirm it is deleted</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#setup">Setup</a></li>
      <li><a href="#search-ncbi">Search NCBI</a></li>
      <li><a href="#retrieve-sequences">Retrieve sequences</a></li>
      <li><a href="#take-down">Take down</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
